app/controllers/PaypalController.scala

```diff
@@ -46,13 +45,13 @@ class PaypalController(
     val paypalService = paymentServices.paypalServiceFor(request)
     val idUser = IdentityUser.fromRequest(request).map(_.id)
 
-    paypalService.executePayment(paymentId, payerId) match {
-      case Right(_) =>
+    paypalService.executePayment(paymentId, payerId).value flatMap {
+      case Xor.Right(_) =>
```

Comment:
I think this nested pattern match needs to be re-written as it gets hard to follow. I know it's not the point of this PR but that would be nice